userland/examples/tests/gpio_async/README.md

```diff
@@ -0,0 +1,7 @@
+GPIO Async Test App
+===================
+
+This is a basic test application for asynchronous GPIO support. The
+`async_gpio` interface mirrors the `gpio` interface, but is designed for
+uses where there is an asynchronous request needed to control the GPIO, as would
+be common with an I2C GPIO extender.
```

Comment:
I liked the bit from Branden's ADC README where he included some sample output for what it should look like if this test app runs correctly